script/nodenv-install.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+#
+# Install our selected Node version defined in the .node-version file.
+#
+# If our node-build version is outdated and it can't build the version we want
+# then we try upgrading node-build and installing again.
+
+if nodenv install --skip-existing; then
+  echo "Correct Node version is installed."
+else
+  echo "Upgrading node-build:"
+  # If homebrew (macOS) installed, try that first. Otherwise look in plugins directory.
+  brew upgrade node-build || git -C "$(nodenv root)"/plugins/node-build pull
```

Comment:
I had a quick test. It's all good but I didn't like to see:

```
Upgrading node-build:
./script/nodenv-install.sh: line 13: brew: command not found
```

So I have a suggestion:

```suggestion

  if command -v brew &> /dev/null; then
    # Installation via Homebrew is recommended on macOS.
    brew upgrade node-build
  else
    git -C "$(nodenv root)"/plugins/node-build pull
  fi
```

Reply:
Oh, thank you, I should have seen that!

I've not seen `command -v` before. 
I just did a quick comparison with `which` out of interest, and found it performed the same for me (return `0` error code if available and `1` error code if not)

Reply:
Here's some information about `command` being faster and safer: https://stackoverflow.com/a/677212/3377535